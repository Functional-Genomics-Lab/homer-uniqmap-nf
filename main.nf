nextflow.preview.output = true

include { HOMER_GETMAPPABLEREGIONS } from './modules/local/homer/getmappableregions'
include { HOMER_CREATEUNIQMAP      } from './modules/local/homer/createuniqmap'

workflow {
    main:
    ch_versions = Channel.empty()

    // Split FASTA by chromosome
    split_fastas = file(params.fasta).splitFasta(by: 1, file: true)

    // Generate mappable regions
    HOMER_GETMAPPABLEREGIONS(
        split_fastas,
        1000000000,
        50
    )
    ch_versions = ch_versions.mix(HOMER_GETMAPPABLEREGIONS.out.versions)

    // Create uniqmap directory
    HOMER_CREATEUNIQMAP(
        HOMER_GETMAPPABLEREGIONS.out.txt
    )
    ch_versions = ch_versions.mix(HOMER_CREATEUNIQMAP.out.versions)

    publish:
    HOMER_CREATEUNIQMAP.out.uniqmap_dir >> 'homer/uniqmap'
}
