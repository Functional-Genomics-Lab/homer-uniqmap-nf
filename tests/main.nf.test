nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                {
                    assert snapshot(
                        workflow.trace.tasks().size(),
                        path("$outputDir/homer/uniqmap.GRCh38_chr21.50nt.zip"),
                    ).match()
                }
            )
        }
    }
}
