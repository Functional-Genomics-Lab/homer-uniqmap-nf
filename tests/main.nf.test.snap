{
    "Should run without failures": {
        "content": [
            2,
            "uniqmap.GRCh38_chr21.50nt.zip:md5,1988058f42465f701caaf9ee651531dc"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T17:27:39.624901"
    }
}